### ISSUE TITLE
Climate not working with Daikin Altherma EHBX08CB3V

### ISSUE THREAD

**Issue by FSlunkk** [BigFoot2020/daikin_residential_brp069a62#6]
Hello,
I have tested this integration on a system with my Alterma EHBX08CB3V and a BRP069a62. Climate don't work and i don't know how fix it.
Could I have some help?
Thanks in advance
Fabien
![PAC](https://user-images.githubusercontent.com/99389346/162729740-f4c2b0ef-1c6a-40a7-a21c-e99aa34f1257.png)
There is the log file :
Logger: homeassistant.components.climate
Source: custom_components/daikin_residential_brp069a62/daikin_base.py:181
Integration: Thermostat (documentation, issues)
First occurred: 13:11:11 (2 occurrences)
Last logged: 13:11:11

Error adding entities for domain climate with platform daikin_residential_brp069a62
Error while setting up daikin_residential_brp069a62 platform for climate
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 614, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 799, in add_to_platform_finish
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 532, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 572, in _async_write_ha_state
    attr.update(self.state_attributes or {})
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 261, in state_attributes
    self.current_temperature,
  File "/config/custom_components/daikin_residential_brp069a62/climate.py", line 127, in current_temperature
    return self._device.leavingWater_temperature
  File "/config/custom_components/daikin_residential_brp069a62/daikin_base.py", line 181, in leavingWater_temperature
    return float(self.getValue(ATTR_LW_TEMPERATURE))
TypeError: float() argument must be a string or a number, not 'NoneType'

**Comment by BigFoot2020** [comment]
Hello!

Your device does not support leaving water temperature. you would need to manually edit code and use one of the available temperatures instead. Similar solution has been mentioned in other issuesalready.

To change temperature shown in thermostat card, in HA Config custom_components\daikin_residential_brp069a62\climate.py
Line 127: `return self._device.leavingWater_temperature`
Replace with: `return self._device.inside_temperature`

**Comment by FSlunkk** [comment]
Thanks, it now works.
I just have a problem with the buttons. I can change without worries from hot to cold mode but when I click on the off button, I can no longer reactivate hot or cold mode by clicking on the corresponding button.
Thanks in advance
Fabien

**Comment by BigFoot2020** [comment]
Buttons on the climate entity? Anything relevant in the logs?
Possibly operationMode's for your device are not exactly same, and since these are hardcoded, it does not work correctly.
You should be able to see avaialble operationModes in daikin_puredata.json file HA config folder.

**Comment by FSlunkk** [comment]
Yes, buttons on the climate entity. There is nothing in the log.
Here is an excerpt from the file daikin_puredate.json. Operation Mode's seems to be the same.

                "operationMode": {
                    "settable": true,
                    "value": "heating",
                    "values": [
                        "heating",
                        "cooling",
                        "auto"

**Comment by BigFoot2020** [comment]
Looks like there is a bug in daikin_base.py

Line 278: `if self.getValue(ATTR_CLIMATE_ON_OFF) != ATTR_STATE_OFF:`
Should be: `if self.getValue(ATTR_CLIMATE_ON_OFF) == ATTR_STATE_OFF:`

**Comment by FSlunkk** [comment]
That's right, I fixed it, it works. 
Thanks very much!

**Comment by FSlunkk** [comment]
Hello, By updtating from the version of core 6.5 to 7.5, altherma climate became unavailable. I also update daikin_residential_brp069a62 but it still doesn't work. Do you have an idea to unblock the situation?

**Comment by BigFoot2020** [comment]
Could it still be related to upgrade to Python 3.10 with 2022.7.0, but this should have been fixed by updating this component.
Could you verify that you have latest version?
Specifically in manifest.json
Line 4: `"version": "0.1.1",`
Line 12: `"oic==1.4.0"`
Last one is the actual fix for Python, so if it's not there, add it manually and restart.
If this does not helt, then see if there is anything related in logs.

**Comment by FSlunkk** [comment]
This seems to be the correct version

{
  "domain": "daikin_residential_brp069a62",
  "name": "Daikin Residential Controller for BRP069A62",
  "version": "0.1.1",
  "documentation": "https://github.com/bigfoot2020/daikin_residential_brp069a62",
  "dependencies": [],
  "codeowners": [
    "@bigfoot2020"
  ],
  "issue_tracker": "https://github.com/bigfoot2020/daikin_residential_brp069a62/issues",
  "requirements": [
     "oic==1.4.0"
  ],
  "iot_class": "cloud_polling",
  "config_flow": true
}

**Comment by BigFoot2020** [comment]
Looks correct, so just to cover the basics:
1. Did you restart after updating component?
2. Anything related in logs that might help?

**Comment by FSlunkk** [comment]
Yes, i restart.

There is Logs :
2022-07-15 09:56:58 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration daikin_residential_brp069a62 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-15 09:56:58 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration daikin_residential_altherma which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-15 09:56:58 WARNING (SyncWorker_5) [homeassistant.loader] We found a custom integration daikin_residential which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-15 09:57:50 ERROR (MainThread) [homeassistant.components.water_heater] Error adding entities for domain water_heater with platform daikin_residential_brp069a62
File "/config/custom_components/daikin_residential_brp069a62/water_heater.py", line 134, in min_temp
File "/config/custom_components/daikin_residential_brp069a62/daikin_base.py", line 239, in dhw_target_temperature_minValue
2022-07-15 09:57:51 ERROR (MainThread) [homeassistant.components.water_heater] Error while setting up daikin_residential_brp069a62 platform for water_heater
File "/config/custom_components/daikin_residential_brp069a62/water_heater.py", line 134, in min_temp
File "/config/custom_components/daikin_residential_brp069a62/daikin_base.py", line 239, in dhw_target_temperature_minValue
2022-07-15 09:57:51 ERROR (MainThread) [homeassistant.components.climate] Error adding entities for domain climate with platform daikin_residential_brp069a62
File "/config/custom_components/daikin_residential_brp069a62/climate.py", line 127, in current_temperature
File "/config/custom_components/daikin_residential_brp069a62/daikin_base.py", line 181, in leavingWater_temperature
2022-07-15 09:57:52 ERROR (MainThread) [homeassistant.components.climate] Error while setting up daikin_residential_brp069a62 platform for climate
File "/config/custom_components/daikin_residential_brp069a62/climate.py", line 127, in current_temperature
File "/config/custom_components/daikin_residential_brp069a62/daikin_base.py", line 181, in leavingWater_temperature

**Comment by BigFoot2020** [comment]
ok, so updating most likely reverted your manual change. You will neeed to set this back:
In HA Config custom_components\daikin_residential_brp069a62\climate.py
Line 127: `return self._device.leavingWater_temperature`
Replace with: `return self._device.inside_temperature`

**Comment by FSlunkk** [comment]
Yes, that was it. I did not think about it. Thanks a lot

**Comment by FSlunkk** [comment]
Hello, i have a new little problem. The setpoint temperature change is no longer taken into account. I have this error message appearing in the log :

Logger: custom_components.daikin_residential_brp069a62.device
Source: custom_components/daikin_residential_brp069a62/device.py:317
Integration: Daikin Residential Controller for BRP069A62 (documentation, issues)
First occurred: 10:26:07 (3 occurrences)
Last logged: 10:26:09

FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)


Could you help me? 
Thanks in advance
Fabien

**Comment by BigFoot2020** [comment]
Could you copy error message from full log?

**Comment by FSlunkk** [comment]
2022-11-08 08:49:36.578 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration daikin_residential_altherma which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-11-08 08:49:36.603 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration daikin_residential_brp069a62 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-11-08 08:49:36.616 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration daikin_residential which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-11-08 08:50:40.898 ERROR (MainThread) [homeassistant.components.water_heater] Error adding entities for domain water_heater with platform daikin_residential_brp069a62
File "/config/custom_components/daikin_residential_brp069a62/water_heater.py", line 134, in min_temp
File "/config/custom_components/daikin_residential_brp069a62/daikin_base.py", line 239, in dhw_target_temperature_minValue
2022-11-08 08:50:41.764 ERROR (MainThread) [homeassistant.components.water_heater] Error while setting up daikin_residential_brp069a62 platform for water_heater
File "/config/custom_components/daikin_residential_brp069a62/water_heater.py", line 134, in min_temp
File "/config/custom_components/daikin_residential_brp069a62/daikin_base.py", line 239, in dhw_target_temperature_minValue
2022-11-08 10:26:07.741 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)
2022-11-08 10:26:07.777 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)
2022-11-08 10:26:09.393 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)
2022-11-08 12:59:04.177 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)
2022-11-08 12:59:20.061 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)
2022-11-08 12:59:20.681 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)
2022-11-08 12:59:21.586 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)
2022-11-08 12:59:23.398 ERROR (MainThread) [custom_components.daikin_residential_brp069a62.device] FAILED to validate set_data params: Type of value (<class 'float'>) is not the expected type (<class 'int'>)

**Comment by BigFoot2020** [comment]
Ok, this was general problem with stepValue validation. Not sure how it worked before, but it was anyway not properly implemented, so I removed it for now and updated file in GIT.

If you want to test manually (or do not wanto update from GIT) comment out lines 213-220 in device.py

**Comment by FSlunkk** [comment]
I test it manually, it works. Thanks a lot!

**Comment by BigFoot2020** [comment]
In case interested [Daikin Residential for Altherma](https://github.com/speleolontra/daikin_residential_altherma/) now supports BRP069A62. Please try if this would work for your system without manual fixes. If not, please open new issue on that repository.

**Comment by FSlunkk** [comment]
I try it and it works. Thanks!

**Comment by jwillemsen** [comment]
Great! Looks all work the last weeks is working.
